tests/parametric/test_otel_metrics.py

```diff
@@ -0,0 +1,2445 @@
+
+    @pytest.mark.parametrize(
+        ("library_env", "endpoint_env", "test_agent_otlp_grpc_port"),
+        [
+            (
+                {**DEFAULT_ENVVARS},
+                "OTEL_EXPORTER_OTLP_ENDPOINT",
+                4320,
+            ),
+        ],
+    )
+    def test_otlp_custom_endpoint(
+        self, library_env, endpoint_env, test_agent_otlp_grpc_port, otlp_endpoint_library_env, test_agent, test_library
+    ):
+        """Metrics are exported to custom OTLP endpoint."""
+        name = "test_otlp_custom_endpoint-counter"
+        with test_library as t:
+            generate_default_counter_data_point(t, name)
+
+        assert (
+            urlparse(library_env[endpoint_env]).port == 4320
+        ), f"Expected port 4320 in {urlparse(library_env[endpoint_env])}"
+
+        metrics = test_agent.wait_for_num_otlp_metrics(num=1)
+        scope_metrics = metrics[0]["resource_metrics"][0]["scope_metrics"]
+        assert scope_metrics is not None
+
+    @pytest.mark.parametrize(
+        ("library_env", "endpoint_env", "test_agent_otlp_grpc_port"),
+        [
+            (
+                {**DEFAULT_ENVVARS},
+                "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+                4321,
+            ),
+        ],
+    )
+    def test_otlp_metrics_custom_endpoint(
+        self, library_env, endpoint_env, test_agent_otlp_grpc_port, otlp_endpoint_library_env, test_agent, test_library
+    ):
+        """Metrics are exported to custom OTLP endpoint."""
+        name = "test_otlp_metrics_custom_endpoint-counter"
+        with test_library as t:
+            generate_default_counter_data_point(t, name)
+
+        assert (
+            urlparse(library_env[endpoint_env]).port == 4321
+        ), f"Expected port 4321 in {urlparse(library_env[endpoint_env])}"
+
+        metrics = test_agent.wait_for_num_otlp_metrics(num=1)
+        scope_metrics = metrics[0]["resource_metrics"][0]["scope_metrics"]
+        assert scope_metrics is not None
+
+
+@features.otel_metrics_api
+@scenarios.parametric
+class Test_Otel_Metrics_Configuration_OTLP_Exporter_Metrics_Headers:
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                **DEFAULT_ENVVARS,
+                "OTEL_EXPORTER_OTLP_HEADERS": "api-key=key,other-config-value=value",
+                "OTEL_EXPORTER_OTLP_PROTOCOL": "http/protobuf",
+            },
+        ],
+    )
+    def test_custom_http_headers_included_in_otlp_export(self, test_agent, test_library, library_env):
+        """OTLP metrics are emitted when enabled."""
+
+        name = "test_custom_http_headers_included_in_otlp_export-counter"
+        with test_library as t:
+            generate_default_counter_data_point(t, name)
+
+        metrics = test_agent.wait_for_num_otlp_metrics(num=1)
+        scope_metrics = metrics[0]["resource_metrics"][0]["scope_metrics"]
+        assert scope_metrics is not None
+
+        requests = test_agent.requests()
+        test_agent.clear()
```

Comment:
Do we need this operation? We spin up a new instance of the testagent for each test run.